### DISCUSSION TITLE
REQ: LINUX CLAP versions of your plugins

### DISCUSSION THREAD

**Discussion by ghost** [ryukau/VSTPlugins#53] [Ideas]
Hi,

Is there any porting of these **beautiful** plugins to the CLAP format (_Linux_)?

Greetings

**Comment by ryukau** [comment]
No. At least I'm not aware of. Also I'm not planning to work on it in near future.

You could port the plugins by yourself. It's likely faster. A while ago, I prepared [a repository using JUCE](https://github.com/ryukau/UhhyouPluginsJuce) to test plugin formats other than VST 3. That JUCE repository is structured to mostly match the code used in this repository. I guess the rewriting can be done mechanically, but it may require to parse C++.

I stopped working on JUCE because the DAW I'm using (FL Studio) is still catching up the latest features around plugin formats and MIDI 2. It's pointless if I can't compare the features between different formats. Also an apparent downside is that the compilation takes longer compared to this repository.

I'll keep opening this issue.

**Reply by ghost** [reply]
> No. At least I'm not aware of. Also I'm not planning to work on it in near future.
> 
> You could port the plugins by yourself. It's likely faster. A while ago, I prepared [a repository using JUCE](https://github.com/ryukau/UhhyouPluginsJuce) to test plugin formats other than VST 3. That JUCE repository is structured to mostly match the code used in this repository. I guess the rewriting can be done mechanically, but it may require to parse C++.
> 
> I stopped working on JUCE because the DAW I'm using (FL Studio) is still catching up the latest features around plugin formats and MIDI 2. It's pointless if I can't compare the features between different formats. Also an apparent downside is that the compilation takes longer compared to this repository.
> 
> I'll keep opening this issue.

Hello and thanks for your answer, I understand (via Linux I use Reaper and Bitwig) I don't know anything about FL. 
Let me thank you anyway for your excellent plugins! And for the fact that you make them available for Linux users as well! 
I encourage you because you're good, and your plugins are truly inspiring!

Greetings :-)

**Comment by ryukau** [comment]
This is a follow up on #54.

I looked into CLAP examples and it's not looking good for casual use. CLAP is just a format, and it's lacking a GUI library. On this aspect, it's worse than VST 3 because VSTGUI works out of the box at least on Windows and macOS. For now, if I write new plugins in CLAP, I can only support Windows.

I tried to build example clap plugins in the link below. It worked on my Windows environment, however failed to build on macOS (M2, Sequoia 15.2). So this example can't be used for now.

[free-audio/clap-plugins](https://github.com/free-audio/clap-plugins)

There's another example using imgui. This one is lacking Linux support.

[free-audio/clap-saw-demo-imgui: Demo Synth for CLAP using ImGui](https://github.com/free-audio/clap-saw-demo-imgui)
